src/components/pages/devices/flyouts/deviceDelete/deviceDelete.js

```diff
@@ -61,13 +67,49 @@ export class DeviceDelete extends Component {
     if (this.subscription) this.subscription.unsubscribe();
   }
 
+  /*
+  Because the ag-grid doesn't update its state for selected items when something is removed from its props/rowData,
```

Comment:
Check that this is still true with the changes to the data model that went in yesterday.